transactions.go

```diff
@@ -0,0 +1,58 @@
+package breez
+
+import (
+	"context"
+	"github.com/breez/lightninglib/lnrpc"
+	"io"
+	"time"
+)
+
+func watchSwapTransactions() {
+	for {
+		err := subscribeTransactionsOnce()
+		if err != nil {
+			log.Errorf("Error trying to watchSwapTransactions:", err)
+		}
+		time.Sleep(5 * time.Second)
+	}
+}
+
+func subscribeTransactionsOnce() error {
+	transactionStream, err := lightningClient.SubscribeTransactions(context.Background(), &lnrpc.GetTransactionsRequest{})
+	if err != nil {
+		return err
+	}
+	for {
+		log.Infof("new Recv call")
+		t, err := transactionStream.Recv()
+		if err == io.EOF {
+			log.Infof("Stream stopped. Need to re-registser")
+			break
+		}
+		if err != nil {
+			log.Infof("Error in stream: %v", err)
+			return err
+		}
+
+		if t.NumConfirmations > 0 {
+			fetchSwapAddresses(func(addr *swapAddressInfo) bool {
+				for _, a := range t.DestAddresses {
+					if a == addr.Address {
+						sendSwapInvoice(t.DestAddresses[0], t.TxHash, t.Amount, addr.PaymentPreimage)
```

Comment:
If the sendSwapInvoice fails it won't retry. We need to think of a reliable way to send these invoices and get payment.
One options is to split it to two parts:
1. On transaction received just create the payment request and save it on the address, this way it will be done once.
2. We already have the function getPaymentsForConfirmedTransactions, that is called from the mobile where needed. You can call it from here as well after attached the payment request to the address.
3. Also need to convert the getPaymentsForConfirmedTransactions to fetch the payment request from the address into the grpc parameter.